### DISCUSSION TITLE
Patreon

### DISCUSSION THREAD

**Discussion by calonmerc** [nebhead/PiFire#127] [General]
Besides actually jumping into code every now and then, I'd love to be able to throw some money at you, recurrently, to continue developing this and keeping it alive! (I know there's a PayPal, but it's a one time thing and I'd like to be able to just send you $10 every month)

**Comment by nebhead** [comment]
Oh man, that is super kind of you, but it's cool - I will keep on maintaining this thing for the foreseeable future. :) Your contribution was payment enough for me.  I learn a lot from those that help contribute and it's awesome.  If you want to chip in a few extra bucks, that's cool too, but no need to make it an ongoing thing.  I'll keep the lights on.

**Reply by calonmerc** [reply]
I definitely plan on adding some more contributions! But I'm a big proponent of fiscally supporting open source projects I love! $10/month from me isn't much, but it might be enough you can explore additional hardware support.

I also want to look at building a PiFire image for Raspberry Pi; similar to OctoPi for OctoPrint! A relatively easy thing in GitHub actions, but also helps to have a separate mirror (I own a small hosting company 😅)

**Reply by nebhead** [reply]
> I definitely plan on adding some more contributions! But I'm a big proponent of fiscally supporting open source projects I love! $10/month from me isn't much, but it might be enough you can explore additional hardware support.

That's awesome, I also try to kick back to all of the Open Source projects that I use with some funding.  Anyway, just don't want anyone to feel obligated to send money, as this is just a personal hobby project right now.  Maybe someday it could be bigger - but I do also have a day job. :P 

> I also want to look at building a PiFire image for Raspberry Pi; similar to OctoPi for OctoPrint! A relatively easy thing in GitHub actions, but also helps to have a separate mirror (I own a small hosting company sweat_smile)

I had definitely given this some thought, but I have never built (or released) an image before.  I would love to learn how to do this.  I believe @weberbox may have been experimenting with this, using Diet-Pi, but I think it would certainly be cool to have the option to do this in addition to the current install method.  

Speaking of hosting, since you mentioned it, I had thoughts of setting up a PiFire online service that connects with users PiFire instances to do things like save cookfiles, save/backup settings, share recipes, etc... but hadn't really gone as far as looking for a host that could run a Flask based site.  Thoughts on this?

**Reply by calonmerc** [reply]
> Anyway, just don't want anyone to feel obligated to send money, as this is just a personal hobby project right now.

Sure, it's a personal project, that's now also my (and appears many others) personal project including all your hard work. As long as you're working on it and I'm benefitting from it, it's my pleasure to do it! I don't think having a Patreon, Kofi, or other gives much of an obligation to anyone who wouldn't want to give anyways. 

CustomPiOS by Guy Sheffer is what OctoPi is based on, it works with Github actions to build the images, then you can push the images to different locations or just leave them as artifacts in GH. If you're good with it, I'd love to invest an afternoon or two getting that started! 

As per hosting, I know theoretically my services can do Flask/Python/etc, but never actually done it. I know _most_ shared hosting is compatible with Python in some manor or another.

**Reply by nebhead** [reply]
> > Anyway, just don't want anyone to feel obligated to send money, as this is just a personal hobby project right now.
> 
> Sure, it's a personal project, that's now also my (and appears many others) personal project including all your hard work. As long as you're working on it and I'm benefitting from it, it's my pleasure to do it! I don't think having a Patreon, Kofi, or other gives much of an obligation to anyone who wouldn't want to give anyways.

Fair enough!  I think you are starting to wear me down.  Perhaps I will look into setting up Patreon or Kofi and see what that entails. :smile: 

> CustomPiOS by Guy Sheffer is what OctoPi is based on, it works with Github actions to build the images, then you can push the images to different locations or just leave them as artifacts in GH. If you're good with it, I'd love to invest an afternoon or two getting that started!

Very cool.  Yeah, if you don't mind, it would be great to learn more about how this works and see if it can be applied to this project.  I'm curious how things would work, like setting up the wifi, locale, default user account, etc. so that on first boot the system is ready to be setup. 

> As per hosting, I know theoretically my services can do Flask/Python/etc, but never actually done it. I know _most_ shared hosting is compatible with Python in some manor or another.

Yeah, I was looking at Linode to get started, and maybe just play around with.  But mostly because I see them advertised by youtubers so often.  I'm sure there are other good options, like going with the big guys like AWS, Azure or Google.  At this point, I kind of just want to start messing around with something and learning, which I can do on my local machine.  Some day, when things are ready, I'll figure out what hosting makes sense and give it a go.

**Reply by calonmerc** [reply]
As far as I know, Kofi doesn't charge fees to the creator, so might look into that. I'm currently donating to a handful on Patreon and I want to eventually switch on all that have both :)

I'll spend some time working out the basics of the install script for CustomPiOS, would love to chat with you about some things first though! (Will probably make a new discussion for it)

I'd gladly set you up a small VPS if you wanted, not going to be any of a burden on my system!

**Reply by nebhead** [reply]
> I'll spend some time working out the basics of the install script for CustomPiOS, would love to chat with you about some things first though! (Will probably make a new discussion for it)

Cool!  Yeah, let's setup a new discussion (I haven't setup anything on discord yet) and we can hash out details there. 
 
> I'd gladly set you up a small VPS if you wanted, not going to be any of a burden on my system!

Thank you!  That would be super cool.  I'll maybe reach out to you when I'm ready.  Still hacking on some core features, not quite ready to start on that project yet.

**Reply by weberbox** [reply]
I use Diet-pi virtual machines in windows hyper-v. This is just for development and debugging as it is much easier than dealing with a physical pi and way faster. Usually towards the end of a feature development etc I will start using a pi just to verify compatibility etc. 

I think an image would be pretty cool for pifire. It might make getting up and running way faster/easier for new people. 

Something to consider would be having a flasher like the official raspbian one where you can enter a wifi password etc then it flashes the pifire image.

I had worked on creating a .deb package for an easier install but maybe an image would be even easier.

**Comment by calonmerc** [comment]
On a follow up thought, I don't think you should worry about taking money from people if they genuinely want to. You're still offering everything for free! Imagine it as "you built the pellet controller I wanted, but I can't have you over every month to enjoy some of the delicious products that come from it, so have a beer (or coffee) or two on me"

Plus, I think you might be undervaluing the work, look at rpilocator's latest tweet, immediately made me think of you!!
[Screenshot_2022-10-05-12-32-50-94_0b2fce7a16bf2b728d6ffa28c8d60efb.jpg](https://user-images.githubusercontent.com/6835832/194125038-20b3f692-02d1-4124-99e0-0ee0a6fd1600.jpg)

https://twitter.com/rpilocator/status/1577712967319261192?t=4wJWZOj6eJc3lbMqfF90iA&s=19
